pkg/internal/util/cli/status.go

```diff
@@ -34,24 +34,28 @@ type Status struct {
 	logger  log.Logger
 }
 
-// StatusForLogger returns a status object for the logger l,
+// StatusForLogger returns a new status object for the logger l,
 // if l is the kind default logger it will inject a wrapped writer into it
 // and if we've already attached one it will return the previous status
 func StatusForLogger(l log.Logger) *Status {
 	s := &Status{
 		logger: l,
 	}
 	if v, ok := l.(*logger.Default); ok {
```

Comment:
the CLI should probably explicitly setup a spinner capable logger so this logger can otherwise be used without this stuff. I'll make the logger nicer in a future commit to keep this PR from being massive